File: src/algorithm/template/detection/base/interpretation_drawing.py
from abc import abstractmethod
from typing import Never

from src.algorithm.template.detection.base.interpretation import FieldInterpretation


class FieldInterpretationDrawing:
    def __init__(
        self,
        field_interpretation: FieldInterpretation,
    ) -> None:
        self.field_interpretation = field_interpretation
        self.tuning_config = field_interpretation.tuning_config
        self.field = field_interpretation.field

    @abstractmethod
    def draw_field_interpretation(
        self, marked_image, image_type, evaluation_meta, evaluation_config_for_response
    ) -> Never:
        msg = "Not implemented"
        raise Exception(msg)

File: omrchecker-js/packages/core/src/processors/detection/base/interpretationDrawing.ts
/**
 * Base interpretation drawing class.
 *
 * TypeScript port of src/processors/detection/base/interpretation_drawing.py
 * Maintains 1:1 correspondence with Python implementation.
 */

import cv from '../../../utils/opencv';
import type { FieldInterpretation } from './interpretation';

/**
 * Abstract base class for field interpretation drawing.
 *
 * Provides interface for drawing field interpretations on marked images.
 */
export abstract class FieldInterpretationDrawing {
  public fieldInterpretation: FieldInterpretation;
  public tuningConfig: Record<string, unknown>;
  public field: FieldInterpretation['field'];

  constructor(fieldInterpretation: FieldInterpretation) {
    this.fieldInterpretation = fieldInterpretation;
    this.tuningConfig = fieldInterpretation.tuningConfig;
    this.field = fieldInterpretation.field;
  }

  /**
   * Abstract method to draw field interpretation.
   *
   * Must be implemented by subclasses.
   *
   * @param markedImage - Image to draw on
   * @param imageType - Image type ('GRAYSCALE' or 'COLORED')
   * @param evaluationMeta - Evaluation metadata
   * @param evaluationConfigForResponse - Evaluation configuration
   */
  abstract drawFieldInterpretation(
    markedImage: cv.Mat,
    imageType: 'GRAYSCALE' | 'COLORED',
    evaluationMeta?: any,
    evaluationConfigForResponse?: any
  ): void;
}

